src/widgets/programs/programs.resource.tsx

```diff
@@ -1,29 +1,43 @@
 import { openmrsObservableFetch, openmrsFetch } from "@openmrs/esm-api";
-import { Observable, of } from "rxjs";
-import { map, take } from "rxjs/operators";
+import { Observable } from "rxjs";
+import { map } from "rxjs/operators";
+import { FetchResponse } from "@openmrs/esm-api/dist/openmrs-fetch";
+import { PatientProgram, Program, LocationData, SessionData } from "../types";
 
 export function fetchEnrolledPrograms(
   patientID: string
-): Observable<PatientProgram[]> {
-  return openmrsObservableFetch(
+): Observable<Array<PatientProgram>> {
+  return openmrsObservableFetch<Array<PatientProgram>>(
     `/ws/rest/v1/programenrollment?patient=${patientID}`
   ).pipe(
-    map(
-      ({ data }) =>
-        data["results"].sort((a, b) =>
-          b.dateEnrolled > a.dateEnrolled ? 1 : -1
-        ),
-      take(3)
+    map(({ data }: { data: Array<PatientProgram> }) => data["results"]),
+    map((results: Array<PatientProgram>) =>
+      results.sort((a, b) => (b.dateEnrolled > a.dateEnrolled ? 1 : -1))
+    )
+  );
+}
+
+export function fetchActiveEnrollments(
+  patientID: string
+): Observable<Array<PatientProgram>> {
+  return openmrsObservableFetch<Array<PatientProgram>>(
+    `/ws/rest/v1/programenrollment?patient=${patientID}`
+  ).pipe(
+    map(({ data }: { data: Array<PatientProgram> }) => data["results"]),
+    map((results: Array<PatientProgram>) =>
+      results
+        .filter(res => !res.dateCompleted)
+        .sort((a, b) => (b.dateEnrolled > a.dateEnrolled ? 1 : -1))
     )
   );
 }
 
 export function getPatientProgramByUuid(
   programUuid: string
 ): Observable<PatientProgram> {
```

Comment:
Duplicated - I would only have the type implicitly in the `return` statement if possible.